modules/apps/change-tracking/change-tracking-web/src/main/resources/META-INF/resources/publications/js/components/PublicationTimeline.js

```diff
@@ -44,16 +118,29 @@ const PublicationTimeline = ({timelineItems}) => {
 								<ClayLayout.ContentCol>
 									<TimelineDropdownMenu
 										deleteURL={
-											timelineItem.dropdownMenu.deleteURL
+											timelineItem.status.code ===
+											WORKFLOW_STATUS_DRAFT
+												? timelineItem.actions.delete
+														.href
+												: undefined
 										}
 										editURL={
-											timelineItem.dropdownMenu.editURL
+											timelineItem.status.code ===
+											WORKFLOW_STATUS_DRAFT
+												? getEditURL(timelineItem.id)
+												: undefined
 										}
 										revertURL={
-											timelineItem.dropdownMenu.revertURL
+											timelineItem.status.code ===
+											WORKFLOW_STATUS_APPROVED
+												? getRevertURL(timelineItem.id)
+												: undefined
 										}
 										reviewURL={
-											timelineItem.dropdownMenu.reviewURL
+											timelineItem.status.code !==
+											WORKFLOW_STATUS_PENDING
+												? getReviewURL(timelineItem.id)
+												: undefined
```

Comment:
These actions used to have permission checks https://github.com/liferay/liferay-portal/blob/master/modules/apps/change-tracking/change-tracking-web/src/main/java/com/liferay/change/tracking/web/internal/timeline/CTCollectionHistoryDataProvider.java#L128-L196. We need to keep those permission checks. I believe `timelineItem.actions` contains the list of available actions for the item, so you should be able to utilize it as a way for permission checking.

Reply:
Looking into making the permission changes. 

I'll double check again with a fresh rebuild again after making the permission changes, but I do see the revert action for `APPROVED` publications on my local (which I'm pretty sure is built from after the recent requested changes). 
![image](https://github.com/liferay-publications/liferay-portal/assets/25592431/6bbc6aae-694a-4f42-bb42-3dd8cb15ab14)

Reply:
Oops, sorry I missed your change. Now I see that you do pass in a url for the revert.